Modules/TPC/src/TH2ReductorTPC.cxx

```diff
@@ -0,0 +1,102 @@
+// or submit itself to any jurisdiction.
+
+///
+/// \file     TH2Reductor.cxx
+/// \author   Marcel Lesch
+/// \author   Cindy Mordasini
+/// \author   Based on the work from Piotr Konopka
+///
+
+#include "QualityControl/QcInfoLogger.h"
+#include "TPC/TH2ReductorTPC.h"
+#include <vector>
+#include <TAxis.h>
+#include <TCanvas.h>
+#include <TH2.h>
+#include <TList.h>
+
+namespace o2::quality_control_modules::tpc
+{
+void TH2ReductorTPC::update(TObject* obj, std::vector<SliceInfo>& reducedSource,
+                            std::vector<std::vector<float>>& axis)
+{
+  // Define the local variables in the default case: 1 single pad
+  // (no multipad canvas, nor slicer), and slicer axes size set to 1 (no slicing).
+  TList* padList = nullptr; // List of TPads if input TCanvas.
+  TH2* histo = nullptr;     // Pointer to the histogram to trend.
+  int numberPads = 1;       // Number of output objects.
+  int axisSize = 1;         // Default value for the main slicer axis.
+  int innerAxisSize = 1;    // Default value for the inner slicer axis.
+
+  // Get the number of pads, and their list in case of an input canvas.
+  const bool isCanvas = (obj->IsA() == TCanvas::Class());
+  if (isCanvas) {
+    auto canvas = static_cast<TCanvas*>(obj);
+    padList = static_cast<TList*>(canvas->GetListOfPrimitives());
+    padList->SetOwner(kTRUE);
+    numberPads = padList->GetEntries();
+  } else {                    // Slicer case: number of pads = axis.size*(innerAxis.size-1).
+    if (axis[0].size() > 1) { // Ensure the slicer config is valid.
+      axisSize = (int)axis.size();
+      innerAxisSize = (int)axis[0].size() - 1;
+    }
+  }
+  ILOG(Info, Support) << "Number of output histograms for the trending of "
+                      << obj->GetName() << ": " << numberPads << ENDM;
+
+  // Access the histograms embedded in 'obj'.
+  for (int iPad = 0; iPad < numberPads; iPad++) {
+    if (isCanvas) {
+      auto pad = static_cast<TPad*>(padList->At(iPad));
+      histo = static_cast<TH2*>(pad->GetListOfPrimitives()->At(0));
+    } else { // 'obj' is already a single histogram.
+      histo = static_cast<TH2*>(obj);
+    }
+
+    if (histo) {
+      // Get the trending quantities defined in 'SlicerInfo'.
+      // The two for-loop do only one pass if we have an input canvas.
+      for (int i = 0; i < axisSize; i++) {
+        for (int j = 0; j < innerAxisSize; j++) {
+          if (!isCanvas && (innerAxisSize > 1)) {
+            histo->GetXaxis()->SetRangeUser(axis[i][j], axis[i][j + 1]);
+          }
+
+          SliceInfo mySlice;
+          mySlice.entries = histo->GetEntries();
+          mySlice.meanX = histo->GetMean(1);
+          mySlice.stddevX = histo->GetStdDev(1);
+          mySlice.errMeanX = mySlice.stddevX / (sqrt(mySlice.entries));
+
+          mySlice.meanY = histo->GetMean(2);
+          mySlice.stddevY = histo->GetStdDev(2);
+          mySlice.errMeanY = mySlice.stddevY / (sqrt(mySlice.entries));
```

Comment:
Add a protection for division by 0. Also add and "else" with the definition of errMeanY = 0.